viz/src/machines/viz.ts

```diff
@@ -1,79 +1,214 @@
                 },
               },
             },
-            "Automatic playing on": {
-              after: {
-                "automatic playing delay": {
-                  actions: "Increment step index",
-                  target: "Automatic playing on",
-                  internal: false,
+            "Managing machine and input execution": {
+              initial: "Idle",
+              states: {
+                Idle: {
+                  on: {
+                    Load: {
+                      target: "Executing machine and input",
+                    },
+                  },
+                },
+                "Executing machine and input": {
+                  entry: [
+                    "Cache input and machine code into context",
+                    "Reset step index",
+                  ],
+                  exit: "Exit loading state from submit button",
+                  type: "parallel",
+                  states: {
+                    "Making request to server": {
+                      initial: "Sending request",
+                      states: {
+                        "Sending request": {
+                          invoke: {
+                            src: "Execute machine and input on server",
+                            onDone: [
+                              {
+                                actions: "Assign machine execution to context",
+                                target: "Received response",
+                              },
+                            ],
+                            onError: [
+                              {
+                                target:
+                                  "#Visualizer.Application is ready.Managing machine and input execution.Failed to execute machine and input",
+                              },
+                            ],
+                          },
+                        },
+                        "Received response": {
+                          type: "final",
+                        },
+                      },
+                    },
+                    "Delaying loading state": {
+                      initial: "Pending",
+                      states: {
+                        Pending: {
+                          after: {
+                            "1000": {
+                              target: "Reached end of delay",
+                            },
+                          },
+                        },
+                        "Reached end of delay": {
+                          type: "final",
+                        },
+                      },
+                    },
+                  },
+                  onDone: {
+                    target: "Fetched machine and input execution",
+                  },
+                },
+                "Fetched machine and input execution": {
+                  entry: "Allow to play execution steps",
+                  on: {
+                    Load: {
```

Comment:
question: it will start the machine execution inside the viz on a load hit ? what about a play or whatever action

Reply:
Ok nevermind, didn't see the entry action `Allow to play execution steps`